### ISSUE TITLE
Pagination in load_as_pandas for large tables

### ISSUE THREAD

**Issue by paulhardman** [delta-io/delta-sharing#114]
The ability to pass limit=n to the delta_sharing.load_as_pandas function is very helpful, but it would be even better if you could also pass start_at=. This would allow large tables to be loaded and processed on a client machine without having to load the whole thing into memory.

eg

delta_sharing.load_as_pandas(table_url, limit=1000) returns rows 0-999
delta_sharing.load_as_pandas(table_url, limit=1000, start_at=1000) returns rows 1000-1999

**Comment by zsxwing** [comment]
Good point. However, it's actually not easy to implement pagination in an efficient way. For example, `delta_sharing.load_as_pandas(table_url, limit=1000, start_at=1000)` will still need to read the first 1000 rows and drop them before returning rows 1000-1999.

**Comment by ckayay** [comment]
Would the filtering work here?

**Comment by zsxwing** [comment]
> Would the filtering work here?

It will still do the filtering in the client side. Can we use `load_as_spark` to load the table as a Spark DataFrame and do the filtering? After that you can convert a Spark DataFrame to a Pandas DataFrame by calling `to_pandas()`. This would make your code more scalable which is what Spark is good at.

**Comment by smmenon** [comment]
@zsxwing I have a similar use case where I am trying to consume the Delta Tables into a UI application. Spark wont be an ideal choice here as it takes some time for spark initialization and hence would be difficult to get the data in sub-seconds or seconds. I am looking for some option to pass filters while load_as_pandas. I could pass a filter so that only the portion that is drawn on my chart ui is loaded. Can this be achieved?
